src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs

```diff
@@ -24,21 +30,34 @@ public class EvmPooledMemory : IEvmMemory
 
         public void SaveWord(in UInt256 location, Span<byte> word)
         {
-            CheckMemoryAccessViolation(in location, WordSize);
-            UpdateSize(in location, WordSize);
+            CheckMemoryAccessViolation(in location, in WordSize256);
+            UpdateSize(in location, in WordSize256);
 
-            if (word.Length < WordSize)
+            int offset = (int)location;
+            if (word.Length == WordSize)
             {
-                Array.Clear(_memory!, (int)location, WordSize - word.Length);
+                // Direct 256bit register copy rather than invoke Memmove
+                Unsafe.WriteUnaligned(
+                    ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(_memory), offset),
+                    Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetReference(word))
+                );
+            }
+            else if (word.Length < WordSize)
+            {
+                Array.Clear(_memory!, offset, WordSize - word.Length);
+                word.CopyTo(_memory.AsSpan(offset + WordSize - word.Length, word.Length));
+            }
```

Comment:
Can this even go this way? I see this comes only from `EvmStack.PopBytes`, maybe that one should return a Vector256 by default and not Span<>? Or we could just ignore this case?

![image](https://github.com/NethermindEth/nethermind/assets/12445221/159ee8ac-e5ed-4af8-b6a6-38050455382f)

Reply:
Done